pytensor/tensor/rewriting/math.py

```diff
@@ -552,6 +552,26 @@ def local_sqrt_sqr(fgraph, node):
         return [new_out]
 
 
+@register_canonicalize
```

Comment:
Thinking a bit more, I would perhaps put this in specialize. This is also when we do other stuff like `pow(x, 0.5) -> sqrt(x)` that we want to catch with this rewrite as well

```suggestion
@register_specialize
```